webdriver/tests/interaction/element_clear.py

```diff
@@ -0,0 +1,133 @@
+from tests.support.asserts import assert_error, assert_success
+from tests.support.inline import inline
+
+
+def clear(session, element):
+    return session.transport.send("POST", "session/{session_id}/element/{element_id}/clear"
+                                  .format(session_id=session.session_id,
+                                          element_id=element.id))
+
+
+# 14.2 Element Clear
+
+def test_no_browsing_context(session, create_window):
+    # 14.2 step 1
+    session.url = inline("<p>This is not an editable paragraph.")
+    element = session.find.css("p", all=False)
+
+    session.window_handle = create_window()
+    session.close()
+
+    response = clear(session, element)
+    assert_error(response, "no such window")
+
+
+def test_element_not_found(session):
+    # 14.2 Step 2
+    response = session.transport.send("POST", "session/{session_id}/element/{element_id}/clear"
+                                      .format(session_id=session.session_id,
+                                              element_id="box1"))
+
+    assert_error(response, "no such element")
+
+
+def test_element_not_editable(session):
+    # 14.2 Step 3
+    session.url = inline("<p>This is not an editable paragraph.")
+
+    element = session.find.css("p", all=False)
```

Comment:
There are a few conditions that makes an element [_editable_](https://w3c.github.io/webdriver/webdriver-spec.html#dfn-editable). You should test all these.

Reply:
This 14.2 Step 3 test is testing if the element is not editable. Did you want me to test all the different elements that are not editable or just add more tests for clearing editable elements in 14.2 Step 8?

Reply:
@andreastt

Reply:
I was thinking that using the P tag as you have done is one equivalency, so we should not need to test all the different HTML elements that do not have contentEditable on them. However, we should test < input > elements that are disabled as well as at least one non-resettable form element like label.